L1Trigger/TrackFindingTMTT/interface/GlobalCacheTMTT.h

```diff
@@ -0,0 +1,56 @@
+#ifndef L1Trigger_TrackFindingTMTT_GlobalCacheTMTT_h
+#define L1Trigger_TrackFindingTMTT_GlobalCacheTMTT_h
+
+// Data shared across all streams by TMTT L1 tracking.
+//
+// Provides the python configuration parameters
+// & optional histogramming/debugging.
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Settings.h"
+#include "L1Trigger/TrackFindingTMTT/interface/HTrphi.h"
+#include "L1Trigger/TrackFindingTMTT/interface/StubWindowSuggest.h"
+#include "L1Trigger/TrackFindingTMTT/interface/TrackerModule.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Histos.h"
+
+#include <list>
+#include <memory>
+
+namespace tmtt {
+
+  class GlobalCacheTMTT {
+  public:
+    GlobalCacheTMTT(const edm::ParameterSet& iConfig)
+        : settings_(iConfig),              // Python configuration params
+          htRphiErrMon_({0., 0, 0, 0}),    // rphi HT error monitoring
+          stubWindowSuggest_(&settings_),  // Recommend FE stub window sizes.
+          hists_(&settings_)               // Histograms
+    {
+      hists_.book();
+    }
+
+    // Get functions
+    Settings& settings() const { return settings_; }
+    HTrphi::ErrorMonitor& htRphiErrMon() const { return htRphiErrMon_; }
+    StubWindowSuggest& stubWindowSuggest() const { return stubWindowSuggest_; }
```

Comment:
Can this be
```suggestion
    const StubWindowSuggest& stubWindowSuggest() const { return stubWindowSuggest_; }
```
?

Ok, the only public non-`const` method
https://github.com/cms-sw/cmssw/blob/de9b0a8d8582aba8fa9c9f221c0e233879735d8e/L1Trigger/TrackFindingTMTT/src/StubWindowSuggest.cc#L22-L23
is protected with a `mutex` so technically it is thread safe. (by the way, the file-scope mutex is too coarse, but maybe this is good-enough for now).